ats/src/atsMachines/slurmProcessorScheduled.py

```diff
@@ -288,149 +287,109 @@ def calculateCommandList(self, test):
-        #  alastor   12         2           6            6 / 2 = 3     12 / 3 = 4
-        #  alastor   12         2           8            8 / 2 = 4     12 / 4 = 3
-        #  alastor   12         2          12           12 / 2 = 6     12 / 6 = 2
-        #  alastor   12         2          16           16 / 2 = 8     12 / 8 = INV (cannot evenly distrubute 16 tasks across 24 cores)
-        #  alastor   12         2          24           24 / 2 = 12    12 /12 = 1
+        # set more srun defaults here, may be over-ridden below
+        # For running within an allocation we set it to allow slurm to use all
+        # the nodes as the default
+        srun_nodes="--nodes=%i-%i" % (1, self.numNodes)
+        srun_partition="--comment=nopartition"
+        srun_cpus_per_task="--cpus-per-task=1"
+
+        # Set the --partition option here
+        # If not within an salloc, then set the partition to be used for the srun line
+        if self.runningWithinSalloc == False:
+            srun_partition="--partition=%s" % self.partition
+
+        # Set the --nodes srun option  here
         #
+        # print("DEBUG SAD JULY num_nodes=%i minNodes=%i self.numNodes=%i " % (num_nodes, minNodes, self.numNodes))
         #
-        # This is all calculated using the "nn=1" ATS option.  If it is present, then we go into
-        # this extra set of calculations and accept or reject the test case.  There is no need for the user
-        # to tell ATS about the number of threads.  If the code can use threads it may do so by specifying command
-        # line arguments.
+        # DEBUG SAD JULY num_nodes=0 minNodes=1 self.numNodes=3
+        
+        # Note these 3 node values are:
+        #   num_nodes     : the nn value for the test, will be 0 if not requested by the user
+        #   minNodes      : the number of nodes calculated based on the np (number of mpi) processes and threads
+        #   self.numNodes : the number of nodes allocated for this run of ATS for all concurren tests, typically 3-6 nodes
+         
+        # we are running on the login node, each test will be a separate srun line and allocation
+        if self.runningWithinSalloc == False:                           
+            if num_nodes > 0:
+                srun_nodes="--nodes=%i-%i" % (num_nodes, num_nodes)     # If user set nn then honor it
+            else:
+                if self.exclusive == True:
+                    test.numNodesToUse = minNodes                       # If user asked for exclusive access to each node
+                    srun_nodes="--nodes=%i-%i" % (minNodes, minNodes)   # then set nodes based on minNodes, which is
+                else:                                                   # determined by the number of processors
+                    nodes="--comment=nonodes"                           # This will allow tests to share nodes with other users
+
+        # we are running on a pre-allocated set of node
+        else:                       
```

Comment:
Optional update: replacing this `else:` with `elif num_nodes > 0:` achieves the same goal and is a bit easier to reason about.

Reply:
will review that David.

Reply:
I made this change